graphene_sqlalchemy/batching.py

```diff
@@ -1,20 +1,28 @@
+from asyncio import get_event_loop
+
 import aiodataloader
 import sqlalchemy
 from sqlalchemy.orm import Session, strategies
 from sqlalchemy.orm.query import QueryContext
 
 from .utils import is_sqlalchemy_version_less_than
 
+# Cache this across `batch_load_fn` calls
+# This is so SQL string generation is cached under-the-hood via `bakery`
+# Caching the relationship loader for each relationship prop.
+RELATIONSHIP_LOADERS_CACHE = {}
```

Comment:
We store the loaders per relationship so each item accesses the same cached result ensuring that the selectin is only triggered once